jcl/src/java.base/share/classes/java/lang/String.java

```diff
@@ -2307,37 +2307,21 @@ public boolean equals(Object object) {
 		return false;
 	}
 
-	private static final void deduplicateStrings(String s1, String s2) {
-		if ((s1.count ^ s2.count) >= 0) {
+	private static final void deduplicateStrings(String s1, Object value1, String s2, Object value2) {
```

Comment:
Should `value1` and `value2` be of type `char[]`?

Also please javadoc this method as well.  Its something we haven't always been good about but should improve that when making changes to method signatures.

Reply:
To give them a more specific type, they would have to be `char[]` or `byte[]` depending on the version using `/*[IF Sidecar19-SE]*/`, `/*[ELSE]*/`. Since the body only uses them at the type `Object` anyway, I wanted to avoid unnecessary noise from the preprocessing directives. However I don't feel super strongly about it and I can make the signature conditional if that's what you prefer.

I'll add the javadoc.

Reply:
good point